src/quick-lint-js/error_decorator.h

```diff
@@ -0,0 +1,54 @@
+// quick-lint-js finds bugs in JavaScript programs.
+// Copyright (C) 2020  Matthew Glazar
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef QUICK_LINT_JS_ERROR_DECORATOR_H
+#define QUICK_LINT_JS_ERROR_DECORATOR_H
+
+#include <quick-lint-js/error.h>
+#include <quick-lint-js/text-error-reporter.h>
+#include <quick-lint-js/vim-qflist-json-error-reporter.h>
+#include <variant>
+
+namespace quick_lint_js {
+class error_decorator : public error_reporter {
```

Comment:
Have you thought about other names for this class? Some ideas from Twitch chat:

* ErrorDecoratorCheckToSeelfErrorFactory
* errorTape
* errno_but_different
* ErrorWriter
* ErrorReader
* error_statistics_collector
* ErrorDelegator
* ErrorSeeker
* ErrorManager
* ErrorTracker
* error_state
* error
* error_status
* ErrorCounter
* Error_God

Reply:
I choose to call it `error_tape` because I like the `errorTape` name but the code is mainly snake case.